components/lock/index.rst

```diff
@@ -0,0 +1,168 @@
+Lock Component
+================
+
+.. seo::
+    :description: Instructions for setting up generic locks in ESPHome.
+    :image: folder-open.svg
+
+The ``lock`` domain includes all platforms that should function like a lock
+with lock/unlock actions.
+
+.. _config-lock:
+
+Base Lock Configuration
+-------------------------
+
+.. code-block:: yaml
+
+    lock:
+      - platform: ...
+        name: "Lock Name"
+
+Configuration variables:
+
+- **name** (**Required**, string): The name of the lock.
+- **icon** (*Optional*, icon): Manually set the icon to use for the
+  lock in the frontend.
+- **inverted** (*Optional*, boolean): Whether to invert the binary
+  state, i.e. report LOCKED states as UNLOCKED and vice versa. Defaults
+  to ``false``.
+- **internal** (*Optional*, boolean): Mark this component as internal. Internal components will
+  not be exposed to the frontend (like Home Assistant). Only specifying an ``id`` without
+  a ``name`` will implicitly set this to true.
+- **on_lock** (*Optional*, :ref:`Action <config-action>`): An automation to perform
+  when the lock is locked. See :ref:`lock-on_lock_unlock_trigger`.
+- **on_unlock** (*Optional*, :ref:`Action <config-action>`): An automation to perform
+  when the lock is unlocked. See :ref:`lock-on_lock_unlock_trigger`..
+- **disabled_by_default** (*Optional*, boolean): If true, then this entity should not be added to any client's frontend,
+  (usually Home Assistant) without the user manually enabling it (via the Home Assistant UI).
+  Defaults to ``false``.
+- **entity_category** (*Optional*, string): The category of the entity.
+  See https://developers.home-assistant.io/docs/core/entity/#generic-properties
+  for a list of available options. Requires Home Assistant 2021.11 or newer.
+  Set to ``""`` to remove the default entity category.
```

Comment:
Not sure if you missed my comment from the other PR. Don't forget to add a `note` at the top of the page to say lock only works with at least HA 2022.3.

https://github.com/esphome/esphome-docs/blob/425e0bc38d35b7945eb086d2403c1a076f89b764/components/number/index.rst?plain=1#L12-L14


```suggestion
- **entity_category** (*Optional*, string): The category of the entity.
  See https://developers.home-assistant.io/docs/core/entity/#generic-properties
  for a list of available options. Set to ``""`` to remove the default entity category.
```